src/main/java/by/soft/testProject/questionnairePortal/service/impl/UserServiceImpl.java

```diff
@@ -0,0 +1,124 @@
+package by.soft.testProject.questionnairePortal.service.impl;
+
+import by.soft.testProject.questionnairePortal.entity.Role;
+import by.soft.testProject.questionnairePortal.entity.User;
+import by.soft.testProject.questionnairePortal.repository.RoleRepository;
+import by.soft.testProject.questionnairePortal.repository.UserRepository;
+import by.soft.testProject.questionnairePortal.service.ServiceException;
+import by.soft.testProject.questionnairePortal.service.UserService;
+import by.soft.testProject.questionnairePortal.service.validator.UserDataValidator;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@Service
+public class UserServiceImpl implements UserService {
+
+    private final UserRepository userRepository;
+    private final RoleRepository roleRepository;
+    private final PasswordEncoder passwordEncoder;
+
+    @Autowired
+    public UserServiceImpl(UserRepository userRepository, RoleRepository roleRepository, PasswordEncoder passwordEncoder) {
+        this.userRepository = userRepository;
+        this.roleRepository = roleRepository;
+        this.passwordEncoder = passwordEncoder;
+    }
+
+    @Override
+    public User register(User user, String role) throws ServiceException {
+        refactorUser(user, role);
+
+        if (userRepository.existsByEmail(user.getEmail())) {
+            throw new ServiceException("Email has already in use");
+        }
+        if (userRepository.existsByPhone(user.getPhone())) {
+            throw new ServiceException("Phone has already in use");
+        }
+
+        return userRepository.save(user);
+    }
+
+    @Override
+    public List<User> getAll() {
+        return userRepository.findAll();
+    }
+
+    @Override
+    public User getByEmail(String email) {
```

Comment:
You can return optional from repository and write something like this: userRepository.getByEmail(email).orElseThrow...
In this case you don't have to check for null while calling this method. Related to all same code.